wandb/sdk/internal/sender.py

```diff
@@ -1497,8 +1506,20 @@ def _send_artifact(
             incremental=artifact.incremental_beta1,
             history_step=history_step,
         )
-
+        if artifact.type == "job" and res is not None:
+            if res["artifactSequence"]["latestArtifact"] is None:
+                self._job_builder._job_version_alias = "v0"
+            elif res["artifactSequence"]["latestArtifact"]["id"] == res["id"]:
+                self._job_builder._job_version_alias = (
+                    f"v{res['artifactSequence']['latestArtifact']['versionIndex']}"
+                )
+            else:
+                self._job_builder._job_version_alias = (
+                    f"v{res['artifactSequence']['latestArtifact']['versionIndex'] + 1}"
+                )
+            self._job_builder._job_seq_id = res["artifactSequence"]["id"]
```

Comment:
I'm a little concerned about this, are we sure the response always has these? Should we wrap this in a try, except to prevent users runs from crashing?

Reply:
also i assumed that when you move these attributes to the job_builder you would also move this logic there...